protocol/x/clob/e2e/permissioned_keys_test.go

```diff
@@ -0,0 +1,561 @@
+package clob_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/types"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	testapp "github.com/dydxprotocol/v4-chain/protocol/testutil/app"
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/constants"
+	testtx "github.com/dydxprotocol/v4-chain/protocol/testutil/tx"
+	aptypes "github.com/dydxprotocol/v4-chain/protocol/x/accountplus/types"
+	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+	"github.com/stretchr/testify/require"
+)
+
+type TestBlockWithMsgs struct {
+	Block uint32
+	Msgs  []TestSdkMsg
+}
+
+type TestSdkMsg struct {
+	Msg            sdk.Msg
+	Authenticators []uint64
+	Fees           sdk.Coins
+	Gas            uint64
+	AccountNum     []uint64
+	SeqNum         []uint64
+	Signers        []cryptotypes.PrivKey
+
+	ExpectedRespCode uint32
+	ExpectedLog      string
+}
+
+func TestPlaceOrder_PermissionedKeys_Failures(t *testing.T) {
+	config := []aptypes.SubAuthenticatorInitData{
+		{
+			Type:   "SignatureVerification",
+			Config: constants.AlicePrivateKey.PubKey().Bytes(),
+		},
+		{
+			Type:   "MessageFilter",
+			Config: []byte("/cosmos.bank.v1beta1.MsgSend"),
+		},
+	}
+	compositeAuthenticatorConfig, err := json.Marshal(config)
+	require.NoError(t, err)
+
+	tests := map[string]struct {
+		smartAccountEnabled bool
+		blocks              []TestBlockWithMsgs
+
+		expectedOrderIdsInMemclob map[clobtypes.OrderId]bool
+	}{
+		"Txn has authenticators specified, but smart account is not enabled": {
+			smartAccountEnabled: false,
+			blocks: []TestBlockWithMsgs{
+				{
+					Block: 2,
+					Msgs: []TestSdkMsg{
+						{
+							Msg: clobtypes.NewMsgPlaceOrder(
+								testapp.MustScaleOrder(
+									constants.Order_Bob_Num0_Id11_Clob1_Buy5_Price40_GTB20,
+									testapp.DefaultGenesis(),
+								),
+							),
+							Authenticators: []uint64{0},
+
+							Fees:       constants.TestFeeCoins_5Cents,
+							Gas:        0,
+							AccountNum: []uint64{1},
+							SeqNum:     []uint64{0},
+							Signers:    []cryptotypes.PrivKey{constants.BobPrivateKey},
+
+							ExpectedRespCode: aptypes.ErrSmartAccountNotActive.ABCICode(),
+							ExpectedLog:      aptypes.ErrSmartAccountNotActive.Error(),
+						},
+					},
+				},
+			},
+			expectedOrderIdsInMemclob: map[clobtypes.OrderId]bool{
+				constants.Order_Bob_Num0_Id11_Clob1_Buy5_Price40_GTB20.OrderId: false,
+			},
+		},
+		"Txn has authenticators specified, but authenticator is not found": {
+			smartAccountEnabled: true,
+			blocks: []TestBlockWithMsgs{
+				{
+					Block: 2,
+					Msgs: []TestSdkMsg{
+						{
+							Msg: clobtypes.NewMsgPlaceOrder(
+								testapp.MustScaleOrder(
+									constants.Order_Bob_Num0_Id11_Clob1_Buy5_Price40_GTB20,
+									testapp.DefaultGenesis(),
+								),
+							),
+							Authenticators: []uint64{0},
```

Comment:
Question (tangential to PR): is it correct that `AuthenticatorId` is [global](https://github.com/dydxprotocol/v4-chain/blob/eefff671ea1c29dfcf5ba174f345821089924e5d/protocol/x/accountplus/keeper/authenticators.go#L172), but authenticators are always stored under account prefix. Is the main motivation behind global ID so that we just need one ID counter?

Reply:
yeah this decision was from x/smart-account, and I think it was mostly just for simplicity (at least from what I've seen in the code)